executor/builder.go

```diff
@@ -762,3 +767,105 @@ func (b *executorBuilder) buildAnalyze(v *plan.Analyze) Executor {
 	}
 	return e
 }
+
+type tableInfo struct {
+	asName    *model.CIStr
+	table     table.Table
+	tableID   int64
+	keepOrder bool
+	desc      bool
+	ranges    []types.IntColumnRange
+}
+
+func (b *executorBuilder) buildTableReader(v *plan.PhysicalTableReader) Executor {
+	ts := &TableReaderExecutor{
+		ctx:    b.ctx,
+		schema: v.Schema(),
+	}
+	ts.dagPB, ts.tableInfo = b.plan2DAGRequest(v.TablePlan)
+	if b.err != nil {
+		return nil
+	}
+	b.err = ts.doRequest()
+	return ts
+}
+
+func (b *executorBuilder) plan2DAGRequest(p plan.PhysicalPlan) (dagReq *tipb.DAGRequest, info *tableInfo) {
+	dagReq = &tipb.DAGRequest{}
+	dagReq.StartTs = b.getStartTS()
+	dagReq.TimeZoneOffset = timeZoneOffset()
+	sc := b.ctx.GetSessionVars().StmtCtx
+	client := b.ctx.GetClient()
+	dagReq.Flags = statementContextToFlags(sc)
+	// Construct executors by v.TablePlan.
+	plans := make([]plan.PhysicalPlan, 0, 5)
+	for {
+		plans = append(plans, p)
+		if len(p.Children()) == 0 {
+			break
+		}
+		p = p.Children()[0].(plan.PhysicalPlan)
+	}
+	for i := len(plans) - 1; i >= 0; i-- {
+		p := plans[i]
+		switch x := p.(type) {
+		case *plan.PhysicalAggregation:
+			aggExec := &tipb.Aggregation{
```

Comment:
This function is too long, extract a method for each physical plan is more clear.
Then pass a single `ctx` argument instead of `sc` and `client`.